web/src/components/DandisetsPage.vue

```diff
@@ -136,14 +136,25 @@ export default {
       return { page, sortOption, sortDir };
     },
     dandisets() {
-      return this.dandisetRequest ? this.dandisetRequest.data : undefined;
+      if (this.girderDandisetRequest === null || this.publishedVersions === null) {
+        return null;
+      }
+      const dandisets = [];
+      for (let i = 0; i < this.totalDandisets; i += 1) {
+        if (this.publishedVersions[i]) {
+          dandisets.push(this.publishedVersions[i]);
+        } else {
+          dandisets.push(this.girderDandisetRequest.data[i]);
+        }
+      }
+      return dandisets;
```

Comment:
TBH I think maps/functional chains are pretty neato, but they're always kinda painful to read. The extra index parameter is cool, but makes it harder to parse. I think that's more a personal weakness of mine.
I definitely try to avoid `for (i=0;i<x;i++)` loops in JS, so I'm doing this, just thought I'd leave my 2¢